src/pages/wallet/wallet-dashboard/wallet-dashboard.ts

```diff
@@ -381,17 +381,19 @@ export class WalletDashboardPage implements OnInit, OnDestroy {
     this.marketDataProvider.refreshPrice();
   }
 
-  private refreshAccount(save: boolean = true) {
-    this.arkApiProvider.api.account.get({ address: this.address }).takeUntil(this.unsubscriber$).subscribe((response) => {
+  private refreshAccount() {
+    this.arkApiProvider.api.account.get({address: this.address}).takeUntil(this.unsubscriber$).subscribe((response) => {
       if (response.success) {
         this.wallet.deserialize(response.account);
-        if (save) { this.saveWallet(); }
+        this.arkApiProvider
```

Comment:
Here we can avoid unnecessary requests: `if (!wallet.isDelegate)`

Reply:
right... it's not possible to unbecome a delegate then right? If so should I also remove the corresponding logic in the `ensureWalletDelegateProperties` method?

Reply:
Yes, please. To make it more readable.

Reply:
done